<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>37-await表达式</title>
  </head>
  <body></body>
  <script>
    async function main() {
      let p = new Promise((resolve, reject) => {
        resolve('OK')
      })
      let errorP = new Promise((resolve, reject) => {
        reject('ERROR')
      })
      // 1. 右侧为promise 的情况
      // let res = await p
      // console.log(res)
      // 2.右侧为其他类型的数据
      // let res2 = await 20
      // console.log(res2)
      // 3. 如果 promise 是失败的状态
      try {
        let res3 = await errorP
      } catch (error) {
        console.log('-------抓到的错误信息-------')
        console.log(error)
      }
    }
    let result = main()
    // console.log(result)
  </script>
</html>
